include/deal.II/base/polynomial_space.h

```diff
@@ -141,7 +141,9 @@ class PolynomialSpace
   void compute (const Point<dim>            &unit_point,
                 std::vector<double>         &values,
                 std::vector<Tensor<1,dim> > &grads,
-                std::vector<Tensor<2,dim> > &grad_grads) const;
+                std::vector<Tensor<2,dim> > &grad_grads = Utilities::EmptyVectorGenerator<Tensor<2,dim> >().get_reference(),
+                std::vector<Tensor<3,dim> > &grad_hessians = Utilities::EmptyVectorGenerator<Tensor<3,dim> >().get_reference(),
+                std::vector<Tensor<4,dim> > &grad_hessian_grads = Utilities::EmptyVectorGenerator<Tensor<4,dim> >().get_reference()) const;
```

Comment:
The lifetime consideration is about the question whether you can even ask the object whether it's empty or not, i.e., whether accessing the length of the vector happens before or after the temporary is destroyed again. In your code, the temporary lives until after the function call is complete, i.e., everything is fine. But I didn't know that off the top of my head, and needed to play with a little toy program to convince myself of it. I think to the average user, this answer is at least not clear. I imagine that even the question is not clear.

My preference would be to break backward compatibility. It's not a function a lot of people will use.